### ISSUE TITLE
Client crash when adding a tome to the system.

### ISSUE THREAD

**Issue by KidMessiah** [Mordenkainen/EquivalentEnergistics#31]
http://pastebin.com/2FJj0fEH
Crash log

**Comment by Mordenkainen** [comment]
Thanks for the report. I'm looking at it now, but probably will not be able to release a fix until this weekend.

**Comment by Mordenkainen** [comment]
A couple of questions:
1. did you change the crystal EMC value in the config?
2. what is the highest EMC value item in your tome?

**Comment by KidMessiah** [comment]
I didn't change the crystal EMC value, but I did set one of the values to the maximum integer it would allow me to. If that's a problem I was only doing it for testing and can revert the changes.

**Comment by Mordenkainen** [comment]
I don't think that should do it.

Do you know what the highest EMC value in your tome is?

Can you load your world or are you dead in the water at the moment?

**Comment by KidMessiah** [comment]
I can load the world if I delete the assembler with the tome in it via mcedit, I will check the highest emc asap but can't for a couple of hours.

**Comment by MaPePeR** [comment]
Because [this line](https://github.com/Mordenkainen/EquivalentEnergistics/blob/master/src/main/java/com/mordenkainen/equivalentenergistics/util/EECraftingPattern.java#L105) initializes `ingredients` with size `9` [the line from the crashlog](https://github.com/Mordenkainen/EquivalentEnergistics/blob/master/src/main/java/com/mordenkainen/equivalentenergistics/util/EMCCraftingPattern.java#L64) runs into an index out of bounds if it needs more than 9 stacks of t2 crystals.

**Comment by Mordenkainen** [comment]
I had forgetten to add any checking to ensure that more than 9 stacks were never generated. Limiting it to 9 is required because of some limits within AE itself,due to use of some MC classes that have a stack limit of 9.

As far as mobius fuel, or other high density EMC items, in general the system wants to work with low EMC items as this produces more accurate ingredient calculations for the autocrafting.

For making a t3, t4, etc.. I can't and maintain ProjectE compatability. Since EMC in PE is an int, a T3 crystal (48.9 Billion EMC) exceeds the maximum EMC allowed on an item by about 46 Billion.

**Comment by KidMessiah** [comment]
I believe octuple compressed cobble is my largest emc.

**Comment by Mordenkainen** [comment]
Ok, thanks. That means it's probably some other failure in the algorithm, that is not due to the EMC being too high.

I am rewriting this now and hope to have an update out this weekend.

**Comment by KidMessiah** [comment]
Oh thank you! I can't wait to play again :3

**Comment by Mordenkainen** [comment]
Oh, also, a new item was created to use as the item form of EMC because any item used will be unable to be crafted by the system. AE does not allow input item = output item in patterns. Had I used Alchemical Coal, Mobius Fuel, etc as my "EMC Storage" items, those items would then be uncraftable via AE.

Instead I chose to add a new item that is not used in any recipes, and can not be crafted by AE. This prevents those issues.

Also, as an end user I think I would be pretty pissed off if I went to pull some Mobius Fuel out of my AE system only to find out that some autocrafting I have set up had "eaten" it all. LOL
